bitcoin/src/blockdata/witness.rs

```diff
@@ -28,14 +28,18 @@ use crate::{Script, VarInt};
 /// saving some allocations.
 ///
 /// [segwit upgrade]: <https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki>
+// Note: serde serialization does not include the varint encoded element length for each element
+// however consensus encoding does include the varint for each element, as does `content`.
```

Comment:
I think this comment would be better placed at impl blocks. Maybe even as doc comment on them explaining that the serialization is equivalent to `Vec<Vec<u8>>` where the inner lengths are not stored. Although it should be kinda obvious since no object in Bitcoin carries it's length inside so why should witness stack items work otherwise?